File: implementation/sipamato/src/test/java/ch/difty/sipamato/persistance/jooq/search/SearchOrderFilterConditionMapperTest.java
package ch.difty.sipamato.persistance.jooq.search;

import static ch.difty.sipamato.db.tables.SearchOrder.SEARCH_ORDER;
import static org.assertj.core.api.Assertions.assertThat;

import org.junit.Test;

import ch.difty.sipamato.db.tables.SearchOrder;
import ch.difty.sipamato.db.tables.records.SearchOrderRecord;
import ch.difty.sipamato.persistance.jooq.FilterConditionMapperTest;
import ch.difty.sipamato.persistance.jooq.GenericFilterConditionMapper;

public class SearchOrderFilterConditionMapperTest extends FilterConditionMapperTest<SearchOrderRecord, ch.difty.sipamato.db.tables.SearchOrder, SearchOrderFilter> {

    private final SearchOrderFilterConditionMapper mapper = new SearchOrderFilterConditionMapper();

    private final SearchOrderFilter filter = new SearchOrderFilter();

    @Override
    protected SearchOrder getTable() {
        return SEARCH_ORDER;
    }

    @Override
    protected GenericFilterConditionMapper<SearchOrderFilter> getMapper() {
        return mapper;
    }

    @Override
    protected SearchOrderFilter getFilter() {
        return filter;
    }

    @Test
    public void creatingWhereCondition_withOwnerIncludingGlobal_searchesForOwnerIdOrGlobal() {
        filter.setOwnerIncludingGlobal(10);
        assertThat(mapper.map(filter).toString()).isEqualTo("(\n  \"PUBLIC\".\"SEARCH_ORDER\".\"OWNER\" = 10\n  or \"PUBLIC\".\"SEARCH_ORDER\".\"GLOBAL\" = true\n)");
    }

    @Test
    public void creatingWhereCondition_withOwner_searchesForOwnerId() {
        filter.setOwner(20);
        assertThat(mapper.map(filter).toString()).isEqualTo("\"PUBLIC\".\"SEARCH_ORDER\".\"OWNER\" = 20");
    }

    @Test
    public void creatingWhereCondition_forGlobal_searchesForGlobal() {
        filter.setGlobal(true);
        assertThat(mapper.map(filter).toString()).isEqualTo("\"PUBLIC\".\"SEARCH_ORDER\".\"GLOBAL\" = true");
    }

    @Test
    public void creatingWhereCondition_forGlobal_searchesForNotGlobal() {
        filter.setGlobal(false);
        assertThat(mapper.map(filter).toString()).isEqualTo("\"PUBLIC\".\"SEARCH_ORDER\".\"GLOBAL\" = false");
    }

}

File: core/core-persistence-jooq/src/test/kotlin/ch/difty/scipamato/core/persistence/search/SearchOrderFilterConditionMapperTest.kt
package ch.difty.scipamato.core.persistence.search

import ch.difty.scipamato.common.persistence.FilterConditionMapperTest
import ch.difty.scipamato.core.db.tables.SearchOrder
import ch.difty.scipamato.core.db.tables.SearchOrder.SEARCH_ORDER
import ch.difty.scipamato.core.db.tables.records.SearchOrderRecord
import ch.difty.scipamato.core.entity.search.SearchOrderFilter
import org.amshove.kluent.shouldBeEqualTo
import org.junit.jupiter.api.Test

internal class SearchOrderFilterConditionMapperTest :
    FilterConditionMapperTest<SearchOrderRecord, SearchOrder, SearchOrderFilter>() {

    override val mapper = SearchOrderFilterConditionMapper()

    override val filter = SearchOrderFilter()

    override val table: SearchOrder = SEARCH_ORDER

    @Test
    fun creatingWhereCondition_withNameMask_searchesForName() {
        filter.nameMask = "fOo"
        mapper.map(filter).toString() shouldBeEqualTo
            """"public"."search_order"."name" ilike (('%' || replace(
                |  replace(
                |    replace('fOo', '!', '!!'),
                |    '%',
                |    '!%'
                |  ),
                |  '_',
                |  '!_'
                |)) || '%') escape '!'""".trimMargin()
    }

    @Test
    fun creatingWhereCondition_withOwnerIncludingGlobal_searchesForOwnerIdOrGlobal() {
        filter.ownerIncludingGlobal = 10
        mapper.map(filter).toString() shouldBeEqualTo
            """(
            |  "public"."search_order"."owner" = 10
            |  or "public"."search_order"."global" = true
            |)""".trimMargin()
    }

    @Test
    fun creatingWhereCondition_withOwner_searchesForOwnerId() {
        filter.owner = 20
        mapper.map(filter).toString() shouldBeEqualTo """"public"."search_order"."owner" = 20"""
    }

    @Test
    fun creatingWhereCondition_forGlobal_searchesForGlobal() {
        filter.global = true
        mapper.map(filter).toString() shouldBeEqualTo """"public"."search_order"."global" = true"""
    }

    @Test
    fun creatingWhereCondition_forGlobal_searchesForNotGlobal() {
        filter.global = false
        mapper.map(filter).toString() shouldBeEqualTo """"public"."search_order"."global" = false"""
    }
}
